### PR TITLE
I think this is right

### PR DESCRIPTION
**Pull request by stewartleslie**

### CODE REVIEW

File: Src/BootCamp.Chapter/BMISLeslie [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            string name;
+            string surname;
+            int age; 
+            decimal height, weight;
+            decimal bmi;
+            string printOut;
+
+            //Console Entries and add value to variables
+            Console.WriteLine("Please enter your first name:");
+            name = (Console.ReadLine());
```
Comment (ignasposka) [inline Src/BootCamp.Chapter/BMISLeslie]: You can declare and assign a variable at once. Like the following: 
`string name = Console.ReadLine()`.

Btw, you don't need to wrap `Console.ReadLine()` into the brackets.

File: Src/BootCamp.Chapter/BMISLeslie [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            Console.WriteLine("Please enter your height in cm:");
+            height = Convert.ToDecimal(Console.ReadLine());
+            Console.WriteLine("Please enter your weight in kg:");
+            weight = Convert.ToDecimal(Console.ReadLine());
+
+            //Work out BMI
+            bmi = weight / ((height / 100) * (height / 100));
+            
+            //Printout
+            printOut = name + " " + surname + " is " + age + " years old, his weight is " + weight + "kg and his height is " + height + "cms. His BMI is " + Math.Round(bmi, 2) + ".";
```
Comment (ignasposka) [inline Src/BootCamp.Chapter/BMISLeslie]: C# has cool feature for this: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

Take a look and try to use in the future 😊

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BMISLeslie.cs  1
         

Clones added
============
- Src/BootCamp.Chapter/BMISLeslie.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845158&bid=16082617)

**Comment by Relem76** [comment]
When did declaring variables on the line that it is initialised? I've always been taught to declare variables at the top as those variables can be used through the whole class.

**Comment by ellmaster** [comment]
You declare it in your function so you can't use it outside of that function. Also you gain no benefit from having them declared outside of the scope of your Main function. I'm a bit nitpicking here though, just to not let bad habits come up.
Simple said:
You gain no benefit by declaring them where they are and therefor you can declare them at place of first usage which makes it more clear to the reader what is happening and why it is happening.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/BMISLeslie.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4845158&bid=16082617)

**Comment by Relem76** [comment]
I did try using float to start with to start with, but couldn't get the BMI calculation to work. And with all the other problems I'm having with GIT, I just went with what works. I do find declaring variables where they are initialised questionable. All code I've read and written have always had them at the top.

**Comment by ellmaster** [comment]
> I did try using float to start with to start with, but couldn't get the BMI calculation to work. And with all the other problems I'm having with GIT, I just went with what works. I do find declaring variables where they are initialised questionable. All code I've read and written have always had them at the top.

https://referencesource.microsoft.com/#mscorlib/system/string.cs,1055
Look from line 1055 onward. You will see that even Microsoft does it the way we tell you to do.

**Comment by Almantask** [comment]
Missing a using statement and so the build is failing, Please fix. Look for hint in errors window.

File: Src/BootCamp.Chapter/BMISLeslie.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+namespace BMISLeslie
```
Comment (ellmaster) [inline Src/BootCamp.Chapter/BMISLeslie.cs]:
```suggestion
using System;
namespace BMISLeslie
```

**Comment by Almantask** [comment]
Expired

**Pull request closed without merging**